ql/src/test/org/apache/hadoop/hive/ql/metadata/StorageHandlerMock.java

```diff
@@ -70,7 +70,7 @@ public class StorageHandlerMock extends DefaultStorageHandler {
     if (writeEntity.getWriteType().equals(WriteEntity.WriteType.INSERT)) {
       return LockType.SHARED_READ;
     }
-    return LockType.SHARED_WRITE;
+    return LockType.EXCLUSIVE;
```

Comment:
Changing the lock means that we are changing the tests. Why is it necessary?

Reply:
This mock is used in 2 test cases:
* testLockingOnInsertIntoNonNativeTables
* testLockingOnInsertOverwriteNonNativeTables

Prior this patch this was ignored in case of insert overwrite and `EXCLUSIVE` was set in case of any type of non transactional table.
After I altered the logic to use the lock type coming from the `storageHandler` the mock returned `SHARED_WRITE` and test `testLockingOnInsertOverwriteNonNativeTables` failed since it still expected `EXCLUSIVE`.

I could alter the assertion in the test but since the default lock type specified by the storage handler is `EXCLUSIVE` I chose altering the mock.
https://github.com/apache/hive/blob/55471330426c2e0a52101c2e535a66f751be76ee/ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveStorageHandler.java#L199-L201

Maybe `super.getLockType` would be better here.